Ginger/Ginger/Run/RunSetsExecutionsHistoryPage.xaml.cs

```diff
@@ -333,40 +416,296 @@ private async void LoadExecutionsHistoryData()
+                    pageNumber = (int)result + (remainingRecords == 0 ? 0 : 1);
+                    break;
+
+                case "nextPage":
+                    if (itemsFetched % recordCount == 0)
+                    {
+                        pageNumber = (itemsFetched / recordCount);
+                    }
+                    else
+                    {
+                        pageNumber = (itemsFetched / recordCount) + 1;
+                    }
+                    break;
+
+                case "previousPage":
+                    pageNumber = (start - 1) / recordCount + 1;
+                    break;
+            }
+
+            return pageNumber;
+        }
+
+        /// <summary>
+        /// Updates the enabled/disabled state of pagination buttons based on current pagination state.
+        /// </summary>
+        private void UpdateButtonStates()
+        {
+            btnFirst.IsEnabled = graphQlClient.HasPreviousPage;
+            btnPrevious.IsEnabled = graphQlClient.HasPreviousPage;
+            btnNext.IsEnabled = graphQlClient.HasNextPage;
+            btnLast.IsEnabled = graphQlClient.HasNextPage;
+            xPageSizeComboBox.IsEnabled = graphQlClient.HasNextPage || graphQlClient.HasPreviousPage;
+        }
+
+        /// <summary>
+        /// Adds the data fetched from GraphQL to the list.
+        /// </summary>
+        private void AddRemoteDataToList(GraphQLResponse<GraphQLRunsetResponse> data)
+        {
+            mExecutionsHistoryList.Clear();
+
+            foreach (var node in data.Data.Runsets.Nodes)
+            {
+                var runSetReport = new RunSetReport
+                {
+                    GUID = node.ExecutionId.ToString(),
+                    RunSetGuid = node.EntityId.Value,
+                    Name = node.Name,
+                    Description = node.Description,
+                    SourceApplication = node.SourceApplication,
+                    SourceApplicationUser = node.SourceApplicationUser,
+                    StartTimeStamp = DateTime.Parse(node.StartTime.ToString(), CultureInfo.InvariantCulture).ToUniversalTime(),
+                    EndTimeStamp = DateTime.Parse(node.EndTime.ToString(), CultureInfo.InvariantCulture).ToUniversalTime(),
+                    Elapsed = node.ElapsedEndTimeStamp,
+                    DataRepMethod = ExecutionLoggerConfiguration.DataRepositoryMethod.Remote
+                };
+
+                if (node.ElapsedEndTimeStamp != null)
+                {
+                    runSetReport.ExecutionDurationHHMMSS = GingerCoreNET.GeneralLib.General.TimeConvert((node.ElapsedEndTimeStamp / 1000).ToString());
+                }
+
+                if (Enum.TryParse<eRunStatus>(node.Status, out eRunStatus runStatus))
+                {
+                    runSetReport.RunSetExecutionStatus = runStatus;
+                }
+                else if (string.Equals(node.Status, "Canceled"))
```

Comment:
Add or update enum with Canceled status so that Enum.TryParse will handle the enum conversion

Reply:
done